Lib/queue.py

```diff
@@ -25,6 +25,15 @@ class Full(Exception):
     pass
 
 
+class ShutDown(Exception):
+    '''Raised when put/get with shut-down queue.'''
+
+
+_queue_alive = "alive"
+_queue_shutdown = "shutdown"
+_queue_shutdown_immediate = "shutdown-immediate"
```

Comment:
I think you can define an enum _QueueState here.

Reply:
See [my comment above](https://github.com/python/cpython/pull/96474#discussion_r1112820110)